### PR TITLE
Endpoint page: improve formatting

### PR DESCRIPTION
**Pull request by paulmelnikow**
Cherry-picked from #2906 (conflicts with that)

Partly addresses #2837 but does not resolve it

### CODE REVIEW

File: frontend/components/endpoint-page.js [hunk 1]
```diff
@@ -34,7 +40,7 @@ const JsonExample = ({ data }) => (
   <JsonExampleBlock>{JSON.stringify(data, undefined, 2)}</JsonExampleBlock>
 )
 JsonExample.propTypes = {
-  data: PropTypes.string.isRequired,
+  data: PropTypes.object.isRequired,
```
Comment (paulmelnikow) [inline frontend/components/endpoint-page.js]: This fixes a runtime proptype error.

**Comment by paulmelnikow** [comment]
This pull request **introduces 1 alert** when merging c1fe91ab677a1c572e16dd136984b668ec9fbf7f into 89ee5d1cb185334e95e97a3d29d18af3f08f3c54 - [view on LGTM.com](https://lgtm.com/projects/g/badges/shields/rev/pr-66659efdbf9315fb678c7e6b9e5a0346e21f1083)

**new alerts:**

* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by paulmelnikow** [comment]
Huh, I'm guessing it's preserving the scroll position after the navigation. Is that happening in #2906 too?

**Comment by calebcartwright** [comment]
Staging app for #2906 isn't loading for me 🤔

**Comment by paulmelnikow** [comment]
I pushed some updates. Will see if it fixes Heroku.

**Pull request merged**